api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -286,4 +286,11 @@ default AppendFiles newFastAppend() {
    * Returns a {@link LocationProvider} to provide locations for new data files.
    */
   LocationProvider locationProvider();
+
+  /**
+   * Return a map of string parameters for this table.
+   *
+   * @return this table's parameters map
+   */
+  Map<String, String> parameters();
```

Comment:
Looks like the addition of `parameters` is for injecting the LF authorized column information. I think we can reduce the scope of this PR a little bit, to first offer only database and table level permission management. 

I understand we cannot leverage `properties` because it will be written back in a commit if we add additional information. But adding another `parameters` is definitely confusing, and we should explore some better ways. It would be much clearer to discuss after we first add database and table level LF support with the existing Iceberg interfaces.

Reply:
Yeah, I really doubt that we should be changing the table API for this.